### PR TITLE
Allow users to configure structured logging which outputs all log messages in JSON format

### LINKED ISSUE #1102: Implement structured logging
We want the possibility to emit structured logs from comit-rs. At the same time, the default log output should be lines to make debugging easier. Structured logging is only useful for environments that need monitoring.

This is to be implemented as follows:

- Change the config table `log_levels` to `logging`
- Add an optional boolean `structured` that defaults to false
- rename the field `comit_node` under `log_levels` to `level`
- Inside `logging.rs`, if `structured` is true, emit a JSON object instead of a single line

The keys of the JSON object should be the one from the `log::Record` (line, path, etc).
We will use it together with `fluentd`, so if there is a recommended format, check it out to see if it makes our life easier if we slightly adapt the JSON that we are emitting.

### PR DESCRIPTION
**Pull request by thomaseizinger**
We removed the log statement about the settings from `main.rs`, please have a look at the relevant commit here ec587fc76e5adb44bcd4be642178773355a8cdb6 for an explanation.

Fixes #1102.

### CODE REVIEW

File: cnd/Cargo.toml [hunk 1]
```diff
@@ -63,6 +63,7 @@ memsocket = "0.1"
 pretty_env_logger = "0.3"
 quickcheck = "0.8.5"
 quickcheck_macros = "0.8.0"
+regex = "1.2.0"
```
Comment (D4nte) [inline cnd/Cargo.toml]: Didn't we say we prefer no to specify the patch version?

Comment (thomaseizinger) [reply cnd/Cargo.toml]: Hmm, that was pre-dependabot though. Does it still matter now?

Comment (D4nte) [reply cnd/Cargo.toml]: I don't know, you're the one who came up with the previous rule...

File: cnd/src/logging/initialize.rs [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+/// 1. `out.finish` only accepts `std::fmt::Arguments` and hence we can only use
+/// `format_args!`
+///
+/// 2. (1) is very likely due to the fact that we don't want to
+/// unnecessarily allocate strings in this function for calls to `Log::log` with
+/// a level that is not even active (think about all the calls to `log::trace`
+/// in all our depenencies that we don't see)
+///
+/// Note that this formatter **CANNOT** handle multiline message as for example
+/// produced by: `log::info!("{:#?}", my_struct);`
```
Comment (D4nte) [inline cnd/src/logging/initialize.rs]: So what's the plan? How do we handle this scenario?

Comment (thomaseizinger) [reply cnd/src/logging/initialize.rs]: I'd propose to just not use these log statements :)

Comment (D4nte) [reply cnd/src/logging/initialize.rs]: Does the compilation fail if we were to use them?

**Pull request merged**